internal/lexer/token.go

```diff
@@ -0,0 +1,51 @@
+package lexer
+
+type TokenType int
+
+const (
+	EOF TokenType = iota
+	IDENTITY
```

Comment:
What is identity?

Reply:
I suggest being more specific here and to have token for PATH (path in our current input source, it may be the current state or anything before the last pipe) and CAPTURER_PATH (path in a specific capture).